doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,312 @@
+{
+    public List<int> Stats { get; set; } = default;
+}
+```
+
+The above types are used in the RPC and Telemetry realizations of Property actions.
+
+## Envoy implementation
+
+Continuing to illustrate in C#, the client-side `PropertyConsumer` is implemented as follows:
+
+```csharp
+public class PropertyWriteRequester<TProp, TStat> : CommandInvoker<TProp, TStat>;
+
+public class PropertyReadRequester<TCtrl, TProp> : CommandInvoker<TCtrl, TProp>;
+
+public class PropertyObserveRequester<TCtrl, TStat> : CommandInvoker<TCtrl, TStat>;
+
+public class PropertyUnobserveRequester<TCtrl, TStat> : CommandInvoker<TCtrl, TStat>;
+
+public class PropertyListener<TProp> : TelemetryReceiver<TProp>;
+```
+
+The service-side `PropertyMaintainer` is implemented as follows:
+
+```csharp
+public class PropertyWriteResponder<TProp, TStat> : CommandExecutor<TProp, TStat>;
+
+public class PropertyReadResponder<TCtrl, TProp> : CommandExecutor<TCtrl, TProp>;
+
+public class PropertyObserveResponder<TCtrl, TStat> : CommandExecutor<TCtrl, TStat>;
+
+public class PropertyUnobserveResponder<TCtrl, TStat> : CommandExecutor<TCtrl, TStat>;
+
+public class PropertyNotifier<TProp> : TelemetrySender<TProp>;
+```
+
+The *write* action is performed by issuing a 'write' Command request.
+
+* The request payload is an instance of `TProp`.
+  * For statically itemized properties, the optional fields in the `TProp` request object contain values for any Properties that are to be written.
+  * For dynamically itemized properties, the keys in the `TProp` request map indicate which Properties to write, and the corresponding values are what is to be written.
+* The response payload is an instance of `TStat`.
+  * For statically itemized properties, each optional field in the `TStat` response object indicates the status of the write operation for the Property named by the field.
+  * For dynamically itemized properties, each value in the `TStat` response map indicates the status of the write operation for the Property named by the key.
+  * The semantics of `TStatVal` values are determined by the service that implements the property.
+
+The *read* action is performed by issuing a 'read' Command request.
+
+* The request payload is an instance of `TCtrl`.
+  * For statically itemized properties, the optional fields in the `TCtrl` request object are `true` for each field whose Property is to be read.
+  * For dynamically itemized properties, the elements in the `TCtrl` request array indicate the names of the Properties to read.
+* The response payload is an instance of `TProp`.
+  * For statically itemized properties, the optional fields in the `TProp` response object contain values for any Properties that have been read.
+  * For dynamically itemized properties, the keys in the `TProp` response map indicate which Properties have been read, and the corresponding values are what was read.
+
+The *observe* action is performed by issuing an 'observe' Command request.
```

Comment:
please add to this doc whether the notification topic would be set as part of this command or determined by the maintainer some other way. (Of course, the envoy would probably set the notification topic to the same thing on every request, similar to how it's done in RPC, but I think it's good to define how the over the wire protocol for this would work)